src/BenchmarkDotNet/Jobs/Argument.cs

```diff
@@ -33,6 +33,20 @@ public MonoArgument(string value)
     [PublicAPI]
     public class MsBuildArgument : Argument
     {
-        public MsBuildArgument(string value) => TextRepresentation = value;
+
+        public MsBuildArgument(string value) => TextRepresentation = EscapeBuildArguments(value);
+
+        /// <summary>
+        /// Escapes the \ with \\ when \ is part of the build argument.
+        /// <see href="https://github.com/dotnet/BenchmarkDotNet/issues/1536">Fixes issue 1536</see>
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        public string EscapeBuildArguments(string args)
+        {
+            if (args.Contains("\\"))
```

Comment:
we should also consider the case where the user provides an already escaped characters like `new MsBuildArgument("/p:a=b\\\\"`) and to be 100% complete a version when there is a mix like `("/p:\a=b\\\\"`)`

Reply:
Added unit test cases to cover the scenarios as suggested